integrations/access/accessrequest/app.go

```diff
@@ -120,6 +135,19 @@ func (a *App) run(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 
+	accessMonitoringRules, err := a.getAllAccessMonitoringRules(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	a.accessMonitoringRules.Lock()
+	for _, amr := range accessMonitoringRules {
+		if !a.amrAppliesToThisPlugin(amr) {
+			continue
+		}
+		a.accessMonitoringRules.rules[amr.GetMetadata().Name] = amr
+	}
+	a.accessMonitoringRules.Unlock()
+
 	process.SpawnCriticalJob(job)
 
 	ok, err := job.WaitReady(ctx)
```

Comment:
I have moved the wait ready above getAllAccessMonitoringRules.
Would locking the cache while waiting for getAllAccessMonitoringRules not achieve the effect of desired effect?
Since the events that arrive in the watcher would end up having to wait until the lock is freed before being updated in the cache?